services/api-server/src/simcore_service_api_server/exceptions/custom_errors.py

```diff
@@ -0,0 +1,17 @@
+from ._base import ApiServerBaseError
+
+
+class CustomBaseError(ApiServerBaseError):
+    pass
+
+
+class InsufficientCreditsError(CustomBaseError):
+    msg_template = "Wallet '{wallet_name}' does not have any credits. Please add some before requesting solver ouputs"
```

Comment:
I would also provide information about the current amount of credits in the wallet, as we do elsewhere with 1WalletNotEnoughCreditsError1, so the user can know how many credits he/she needs to add.